component/tests/common.py

```diff
@@ -231,34 +181,41 @@ def _build_components(self, *classes):
 
 
 class TransactionComponentRegistryCase(common.TransactionCase, ComponentRegistryCase):
-    """Adds Odoo Transaction in the base Component TestCase"""
+    """Adds Odoo Transaction in the base Component TestCase.
 
-    # pylint: disable=W8106
-    def setUp(self):
-        # resolve an inheritance issue (common.TransactionCase does not use
-        # super)
-        common.TransactionCase.setUp(self)
-        ComponentRegistryCase._setup_registry(self)
-        self.collection = self.env["collection.base"]
+    This class doesn't set up the registry for you.
+    You're supposed to explicitly call `_setup_registry` and `_teardown_registry`
+    when you need it, either on setUpClass and tearDownClass or setUp and tearDown.
```

Comment:
Is it supposed to work like that?
In previous versions the registry was set up automatically when using this test class, and now every tests depending on that should explicitly call these methods?
I don't know how it works actually but that seems strange from a user POV. The docstring of `ComponentRegistryCase` says that I could use this class if I want to get the full control of the registry.

Reply:
it will be like this. We don't have/need 2 classes anymore and I find better that each implementation clearly shows where the setup is done. If you want to "hide" or replicate it you define your own common class.

Reply:
@simahawk thanks, so it has been discussed :)